### PR TITLE
React capstone project 

### PR DESCRIPTION
**Pull request by zmekonnen251**
## Description

In this PR, I built a Covid19 Statistics tracker app

### General requirements

- [x] Followed the **Github flow**.
- [x] There are no linter errors.
- [x] The documentation is professional.

### JavaScript requirements

- [x] Followed list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### Project requirements

- [x] Chosed the [Narrativa API](https://covid19tracking.narrativa.com/index_en.html): COVID-19 data Api
- [x] ollow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
  - Colors (select one main color for your website).
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.
- [x] Created Home page and details page
- [x] Created unint and integration tests

Thank you for taking the time to review this pull request.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (aliabbani) [inline README.md:82]: I think we need to add an "Acknowledgements" section or similar in the README file that gives appropriate credit to[ Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the[ author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the[ Creative Commons License](https://creativecommons.org/licenses/).

File: README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+## Built With
+
+- HTML5, CSS3
+- JavaScript ES6, React/Redux, Bootstrap
+- CLI, TDD, gh-pages, Gitflow
+
+## Live demo
+
+[live Demo](covidstatupdate.netlify.app)
```
Comment (aliabbani) [inline README.md]: Let s make sure we added the live demo correctly inside the readme file 😊

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+const App = () => (
+  <Router>
+    <div className="box-border text-white bg-lightpink pb-3 mt-0">
+      <NavBar />
+      <Routes>
+        <Route exact path="/" element={<Main />} />
+      </Routes>
+    </div>
+  </Router>
```
Comment (aliabbani) [inline src/App.js]: I think each page we go for should have its unique path. Please let s add the path to the other 2 pages in the project 😊 .

**Review by aliabbani — CHANGES_REQUESTED** [review]
Hi @zmekonnen251 ✋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
- [x] I cannot find a video link either in your PR summary or in the readme file. Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes. The content of the video should include:
- a description of the project. 
- a demo of the project features.
- you should also highlight some interesting piece of code or something you built that you are very proud of.

- [x] Let's try to follow the[ design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) including the layout composition and space between elements.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by zmekonnen251** [comment]
Here is the loom video Link [Loom link](https://www.loom.com/share/cbda22dcfbe746e79b85620cc764a136)

File: README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
```
Comment (Deepakdanger) [inline README.md]: - [x] Let's add the correct live link of your project, I think it's `https://covidstatupdate.netlify.app/`.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+## 🤝 Contributing
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
```
Comment (Deepakdanger) [inline README.md:79]: - [x] In `Readme` file there should be one `Acknowledgement` section which gives appropriate credit to[ Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the[ author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the[ Creative Commons License](https://creativecommons.org/licenses/). As requested by the previous reviewer.
- **REQUIREMENT** The README contains an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

File: src/App.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
+import Main from './components/Main';
+import NavBar from './components/NavBar';
+
+const App = () => (
+  <Router>
+    <div className="box-border text-white bg-lightpink pb-3 mt-0">
+      <NavBar />
+      <Routes>
+        <Route exact path="/" element={<Main />} />
```
Comment (Deepakdanger) [inline src/App.js]: - [x] As requested by the previous reviewer, let's add the path to the other 2 pages in the project here.
- **REQUIREMENT** The category item detail page has its unique route within the SPA.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @zmekonnen251, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required changes
- [x] _Check the comments under the review._

## Optional changes
You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] In the future when using this project, My suggestion is to get country, continent name through API, with each country and continent pics which help to beautify your project even more. Then you can use this project for interview stuff.
- **Hint**  For getting country, continent name, and pic, there are many free APIs, available on the website, you can use any one of them. This is my only suggestion for future. 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by zmekonnen251** [comment]
Thank you @Deepakdanger  for your helpfull comment and feedback.

File: src/App.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        <Routes>
+          <Route exact path="/" element={<Main />} />
+          <Route
+            exact
+            path="/Details"
+            element={<CountryDetails selectedCountry={selectedCountry || ''} />}
+          />
+        </Routes>
+      </div>
+    </Router>
```
Comment (aliabbani) [inline src/App.js]: As requested by the previous reviewer, let's add the path to the other 2 pages in the project here.
- The category item detail page has its unique route within the SPA. Let's make sure when we navigate(click) to a specific continent, our URL adds its unique path, and when to navigate(click) to a specific country our URL adds its unique path.

**Review by aliabbani — CHANGES_REQUESTED** [review]
Hi @zmekonnen251, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
- [x] Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @zmekonnen251, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

**Pull request merged**